### ISSUE TITLE
k8s-master runs in default namespace

### ISSUE THREAD

**Issue by SaMnCo** [juju-solutions/bundle-canonical-kubernetes#45]
Hi there, 

When you deploy this bundle, 

```
$ kubectl --kubeconfig=kubeconfig get pods -o wide --all-namespaces
NAMESPACE     NAME                     READY     STATUS    RESTARTS   AGE       IP            NODE
default       k8s-master-172.31.55.3   4/4       Running   2          2m        172.31.55.3   172.31.55.3
kube-system   kube-dns-v18-alz7t       3/3       Running   0          2m        10.1.32.3     172.31.43.20
```

I think the master should be in the kube-system namespace, not in default, or at least in a non-default ns, shouldn't it?

**Comment by mbruzek** [comment]
@SaMnCo That looks correct to all the research that I have done. We are setting up the cluster where the master components are not in any special namespace (default) and the dns is in the "kube-system" namespace. This is what we found others doing as well in our research. 

If you wanted to change the namespace of the master components. You can fork the charm/layer and edit the resource definitions for the master components yourself.  

If you can find some definitive best practices to the contrary please include a link here and we will prioritize approprately.

**Comment by chuckbutler** [comment]
I don't think it's incorrect in either case. The functional differences come from what applications resolve in the given namespace (read: DNS)

Sam - why would you want the kubernetes master services isolated in kube-system? Is this intuition based or is this something you've found in documentation elsewhere?

**Comment by SaMnCo** [comment]
TL;DR it is a mix of cosmetic, failsafe and userproofness.

Basically default is where the first "user land" app will start.
When people run kubectl get pods, they only see default.
K8s-master is a control app, the belong to the admin of the cluster. If it
goes away, the whole cluster goes away.
While namespacing does not offer a strong fencing, it just feels safer to
have all the control in a separate NS, and prevent users from harming
themselves by accidentally deleting/damaging something they do not
understand.
If they have to run kubectl get pods --all-namespaces it means they already
know more than 0.
